platform/default/headless_view_egl.cpp

```diff
@@ -4,51 +4,62 @@
 
 #include <cassert>
 
-#include <GL/glx.h>
+#include <GL/gl.h>
+#include <EGL/egl.h>
+#include <EGL/eglext.h>
+#include <gbm.h>
 
 namespace mbgl {
 
 gl::glProc HeadlessView::initializeExtension(const char* name) {
-    return glXGetProcAddress(reinterpret_cast<const GLubyte*>(name));
+    return eglGetProcAddress(name);
 }
 
 void HeadlessView::createContext() {
-    xDisplay = display->xDisplay;
-    fbConfigs = display->fbConfigs;
-
-    if (!glContext) {
-        // Try to create a legacy context
-        glContext = glXCreateNewContext(xDisplay, fbConfigs[0], GLX_RGBA_TYPE, None, True);
-        if (glContext) {
-            if (!glXIsDirect(xDisplay, glContext)) {
-                Log::Error(Event::OpenGL, "failed to create direct OpenGL Legacy context");
-                glXDestroyContext(xDisplay, glContext);
-                glContext = nullptr;
-            }
-        }
+    dpy = display->dpy;
+    config = display->config;
+
+    assert(dpy != EGL_NO_DISPLAY);
+    assert(glContext == EGL_NO_CONTEXT);
+    assert(config != nullptr);
+
+    eglBindAPI(EGL_OPENGL_API);
+
+    glContext = eglCreateContext(dpy, config, EGL_NO_CONTEXT, NULL);
+    if (glContext == EGL_NO_CONTEXT) {
+        mbgl::Log::Error(mbgl::Event::OpenGL, "eglCreateContext() returned error 0x%04x",
+                         eglGetError());
+        throw std::runtime_error("Error creating GL context object");
     }
 
-    if (glContext == nullptr) {
-        throw std::runtime_error("Error creating GL context object.");
+    // TODO(vignatti): think about an optimal size here.
```

Comment:
done.

I actually removed completely the EGL surface code because that's not needed in this single-window case.